File: Algorithms/1297.Maximum-Number-of-Occurrences-of-a-Substring/maximum_number_of_occurrences_of_a_substring.py
class Solution:
    def maxFreq(self, s: str, maxLetters: int, minSize: int, maxSize: int) -> int:
        dic = {}
        length = len(s)
        if length < minSize:
            return 0
      
        res = 0
        for i in range(minSize, length+1):
            for j in range(minSize, maxSize+1):
                if i - j >= 0:
                    tmp = s[i-j: i]
                    if len(set(tmp)) <= maxLetters:
                        if tmp in dic:
                            dic[tmp] += 1
                        else:
                            dic[tmp] = 1
                        res = max(res, dic[tmp])
                else:
                    break
        
      
        return res
            
            

File: Algorithms/1297.Maximum-Number-of-Occurrences-of-a-Substring/maximum_number_of_occurrences_of_a_substring.go
func maxFreq(s string, maxLetters int, minSize int, maxSize int) int {
    length := len(s)
    if length < minSize {
        return 0
    }
    res := 0
    dic := make(map[string]int)
    for i := minSize; i<= length; i++ {
        tmp := s[i-minSize:i]
        
        flag := true
        cnt := make(map[rune]int)
        for _, ch := range tmp {
            _, ok := cnt[ch]
            if !ok {
                cnt[ch] = 1
            }
            if len(cnt) > maxLetters {
                flag = false
                break
            }
	    }

        if flag {
            _, ok := dic[tmp]
            if ok {
                dic[tmp] += 1    
            } else {
                dic[tmp] = 1
            }
            if dic[tmp] > res {
                res = dic[tmp]
            }
        }
    }
    return res
}
